presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -789,4 +794,9 @@ default ConnectorTableProperties getTableProperties(ConnectorSession session, Co
     {
         return Optional.empty();
     }
+
+    default boolean useLegacyInsert()
```

Comment:
Can you add javadoc and explain what is legacy and what is non legacy semantic of inserts? 

Typically, "legacy" means something that we are going to drop some day. Is it a case here as well? I have a feeling that some datasource will always require to pass `null` values directly and may not have support for default values.

Also I would prefer positive name. Maybe `supportDefaultValues`? Why not to use `io.prestosql.spi.connector.ConnectorCapabilities` here?